### PR TITLE
Migrates domains service remote to WordPressKit

### LINKED ISSUE #7309: Migrate DomainsServiceRemote to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #7309 

This PR:
* Adds a new `RemoteDomain` model to the WPKit framework
* Migrates `DomainsServiceRemote.swift` to the WPKit framework
* Adds new unit tests (with stubs) for DomainsServiceRemote.swift to the WPKit framework (`DomainsServiceRemoteRESTTests`)

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app — make sure everything works

Needs review: @astralbodies

### CODE REVIEW

**Comment by bummytime** [comment]
@elibud would you mind taking a look? Thank you!!!

File: WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            let error = error as NSError
+            XCTAssertEqual(error.domain, String(reflecting: WordPressComRestApiError.self), "The error domain should be WordPressComRestApiError")
+            XCTAssertEqual(error.code, WordPressComRestApiError.unknown.rawValue, "The error code should be 7 - unknown")
+            expect.fulfill()
+        })
+        
+        waitForExpectations(timeout: timeout, handler: nil)
+    }
+    
+    func testDeleteSiteWithBadAuthFails() {
```
Comment (elibud) [inline WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift]: Is the name of this test correct? I think it should be `testGetDomainsWithBadAuthFails()`

File: WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            let error = error as NSError
+            XCTAssertEqual(error.domain, String(reflecting: WordPressComRestApiError.self), "The error domain should be WordPressComRestApiError")
+            XCTAssertEqual(error.code, WordPressComRestApiError.authorizationRequired.rawValue, "The error code should be 2 - authorization_required")
+            expect.fulfill()
+        })
+        
+        waitForExpectations(timeout: timeout, handler: nil)
+    }
+
+    func testDeleteSiteWithBadJsonFails() {
```
Comment (elibud) [inline WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift]: Same here.

**Comment by elibud** [comment]
Other than the two small details above this looks good and all test pass 🏅

File: WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift (hunk 1 above)
Comment (bummytime) [inline WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift]: Ugh...yep! Will fix.

File: WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift (hunk 2 above)
Comment (bummytime) [inline WordPressKit/WordPressKitTests/DomainsServiceRemoteRESTTests.swift]: same here! :-)

**Comment by bummytime** [comment]
@elibud Fixed those incorrect test names -  nice catch. 😃

**Comment by bummytime** [comment]
Thank you @elibud!

**Pull request merged**